File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    used = set()
    last_char = words[0][0]

    for i, word in enumerate(words):
        if last_char != word[0] or word in used:
            return [i % n + 1, i // n + 1]
        used.add(word)
        last_char = word[-1]

    return [0, 0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
  const spokenWords = new Set();
  let lastChar = words[0][0];

  for (let i = 0; i < words.length; i++) {
    const currentWord = words[i];
    const player = (i % n) + 1;

    if (currentWord[0] !== lastChar || spokenWords.has(currentWord))
      return [player, Math.floor(i / n) + 1];

    spokenWords.add(currentWord);
    lastChar = currentWord.at(-1);
  }

  return [0, 0];
}
